gateleen-core/src/main/java/org/swisspush/gateleen/core/util/ExpansionDeltaUtil.java

```diff
@@ -277,7 +280,23 @@ public static CollectionResourceContainer verifyCollectionResponse(String target
             throw new ResourceCollectionException("Request did not return data. Invalid usage of params " + params + " ?", StatusCode.BAD_REQUEST);
         }
 
-        JsonObject obj = new JsonObject(data.toString("UTF-8"));
+        // Parse payload
+        final String dataAsString = data.toString(UTF_8);
+        final JsonObject obj;
+        try {
+            obj = new JsonObject(dataAsString);
+        } catch (DecodeException e) {
+            final int MAX_CHARS_TO_PRINT = 4096;
```

Comment:
Why limit?

Reply:
Hmm. Some instinct only.

Should we instead send e.g. 250MB to a client only to tell him, that the response is useless? (Status is 502 and we send some garbage at beginning anyway)

Reply:
Agree. Some limit makes sense.
Though I have the feeling that 4096 is a bit small. Often Stacktraces themselves are 1-2 kBytes in size.
Suggest to increase limit to 64k (65536) - and can be provide the URL from where we got the non-JSON-parseable response?